src/shared/utils/utils.ts

```diff
@@ -214,17 +239,23 @@ export const isObfuscated = (
 };
 
 export const getObfuscatedData = (
-  document: WrappedDocumentV3<OpenAttestationDocumentV3> | WrappedDocumentV2<OpenAttestationDocumentV2>
+  document:
+    | WrappedDocumentV2<OpenAttestationDocumentV2>
+    | WrappedDocumentV3<OpenAttestationDocumentV3>
+    | WrappedDocumentV4<OpenAttestationDocumentV4>
 ): string[] => {
-  if (isWrappedV3Document(document)) {
-    return document.proof.privacy?.obfuscated;
-  }
-
   if (isWrappedV2Document(document)) {
     return document.privacy?.obfuscatedData || [];
+  } else if (isWrappedV3Document(document)) {
+    return document.proof.privacy.obfuscated || [];
+  } else if (isWrappedV4Document(document)) {
+    return document.proof.privacy.obfuscated || [];
   }
 
   throw new Error(
     "Unsupported document type: Can only retrieve obfuscated data from wrapped OpenAttestation v2 & v3 documents."
   );
 };
+
+export const isStringArray = (input: unknown): input is string[] =>
+  Array.isArray(input) && typeof input[0] === "string";
```

Comment:
if array is `["foo", 1]`, it will pass, not sure if this is intended

Reply:
Oh yes, good catch. I can rectify it to:
```suggestion
export const isStringArray = (input: unknown): input is string[] =>
  Array.isArray(input) && input.every((i) => typeof i === "string");
```